### PR TITLE
#30 ADR Hook Concept

### LINKED ISSUE #30: Document Hook Concept in ADR
The ADR is already written but needs a review. This issue is opened afterwarsd to keep track of the work.

### PR DESCRIPTION
**Pull request by Weltraumschaf**

### CODE REVIEW

File: docs/adr/adr_0001.adoc [hunk 1]
```diff
@@ -116,81 +120,81 @@ Keep PersistenceProvider as they are and introduce new "MetaDataProvider" CRD wh
...
-Regarding the Question 1 it seems that both solution approaches are resulting in the same execution model. We descided to implement solution approach 1 and unify both concepts into a more general concept with the name _"hook concept"_. Therefore we exchange the existing name `persistenceProvider` for phase 3 in the excecution model with a more general term `DataProcessing`:
+Regarding question 1 it seems that both solution approaches are resulting in the same execution model. We decided to implement solution approach 1 and unify both concepts into a more general concept with the name _hook concept_. Therefore we exchange the existing name _PersistenceProvider_ for phase 3 in the execution model with a more general term _DataProcessing_:
 
-```
-	┌──────────────────┐          ┌──────────────────┐          ┌──────────────────┐
-	│     Scanning     ├─────────▶│     Parsing      ├─────────▶│ DataProcessing   │
-	│    (Phase 1)     │          │    (Phase 2)     │          │    (Phase 3)     │
-	└──────────────────┘          └──────────────────┘          └──────────────────┘
-```
+// #30 I suggest the term "processing" instead of data processing/
```
Comment (rfelber) [inline docs/adr/adr_0001.adoc]: Good point, lets reduce the term to just "processing" 👍🏻

File: docs/adr/adr_0001.adoc [hunk 2]
```diff
@@ -116,81 +120,81 @@ Keep PersistenceProvider as they are and introduce new "MetaDataProvider" CRD wh
...
+┌──────────────────┐          ┌──────────────────┐          ┌──────────────────┐
+│     scanning     ├─────────▶│     parsing      ├─────────▶│  data processing │
+│    (Phase 1)     │          │    (Phase 2)     │          │    (Phase 3)     │
+└──────────────────┘          └──────────────────┘          └──────────────────┘
+....
 
-Regarding the Question 2 we decided to implement the solution approach 1 with a job-based approach (no active service componend needed).
-The Phase 3 `DataProcessing` will be therefore splitt into to seperate phases named `ReadAndWriteHooks (3.1)` and `ReadOnlyHooks (3.2)`
+Regarding question 2 we decided to implement the solution approach 1 with a job-based approach (no active service component needed). Therefore the phase 3 _data processing_ will be split into two separate phases named _ReadAndWriteHooks_ (3.1) and _ReadOnlyHooks_ (3.2)
+// #30 to what refers 3.1 and 3.2?
```
Comment (rfelber) [inline docs/adr/adr_0001.adoc:191]: The following image details the Phase 3 a bit more into two phases "Processing 3.1: ReadAndWriteHooks" and "Processing 3.2: ReadOnlyHooks" but the numbers are missing there :-/ Can you add them?

**Review by rfelber — CHANGES_REQUESTED** [review]

File: docs/adr/adr_0001.adoc (hunk 2 above)
Comment (Weltraumschaf) [reply docs/adr/adr_0001.adoc:191]: Maybe we should use PlantUML:

- The AsciiArt does not render in all previous monospaced
- w/ images we can add numbers/titles and such easily with Asciidoc.

Comment (Weltraumschaf) [reply docs/adr/adr_0001.adoc:191]: Added it.

File: docs/adr/adr_0001.adoc [hunk 3]
```diff
@@ -16,59 +13,66 @@
...
+* _ScanCompletionHooks_
+* _FindingProcessors_
 
-These could be implemented with a `type` attribute, which declares if they are `ReadOnly` or `ReadAndWrite`.
+These could be implemented with a `type` attribute, which declares if they are *read only* or *read and write*.
 
-The operator would process all these resources in the namespace and execute the `ReadAndWrite` ones first (in serial: one at a time to avoid write conflicts) and then the `ReadOnly` ones (in parallel).
+The _secureCodeBox operator_ would process all these CRDs in the namespace and execute the *read and write* ones first in serial only one at a time to avoid write conflicts and then the *read only* ones in parallel.
+////
+#30: What namespace is meant here?
```
Comment (Weltraumschaf) [inline docs/adr/adr_0001.adoc]: What NS is meant here?

Comment (J12934) [reply docs/adr/adr_0001.adoc]: kubernetes namespace: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/

Comment (Weltraumschaf) [reply docs/adr/adr_0001.adoc]: "in the" implies that it is a special one. Is it the namespace of the operator or which one?

Comment (J12934) [reply docs/adr/adr_0001.adoc]: Ah yeah good note. "In the current namespace" would be better.

Comment (J12934) [reply docs/adr/adr_0001.adoc]: Or "in the kubernetes namespace of the Scan"

Comment (Weltraumschaf) [reply docs/adr/adr_0001.adoc]: Done

**Review by rfelber — APPROVED** [review]

**Pull request merged**